editor/actions.js

```diff
@@ -41,3 +57,42 @@ export function savePost( dispatch, postId, edits ) {
 		} );
 	} );
 }
+
+export function mergeBlocks( dispatch, blockA, blockB ) {
```

Comment:
Similar to what we did with [concatChildren](https://github.com/WordPress/gutenberg/blob/762b618e4ab68165370e658dc3e84a147fca18eb/element/index.js#L80), I wonder if this needs to be necessarily limited to merging two blocks (unless it becomes much more complex).